rabbitmq/src/main/java/io/micronaut/rabbitmq/connect/recovery/TemporarilyDownException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.rabbitmq.connect.recovery;
+
+/**
+ * Indicates that the connection is still down.
+ *
+ * @author Guillermo Calvo
+ * @since 4.1.0
+ */
+public interface TemporarilyDownException {
```

Comment:
annotate with `@FunctionalInterface`

Reply:
I can do that, but please note this interface is not supposed to be _implemented by_ users. Plus, it's meant to decorate actual exception classes. It wouldn't make sense to create a new `TemporarilyDownException` object _on the fly_. In other words, this is not really a _functional interface_ as defined by the Java Language Specification.

@sdelamo Do you really want me to go ahead and annotate it?

```suggestion
@FunctionalInterface
public interface TemporarilyDownException {
```

Reply:
Yes, I think we should annotate it since it contains one abstract method.